app/services/taxonomy/taxonomy_query.rb

```diff
@@ -0,0 +1,39 @@
+module Taxonomy
+  class TaxonomyQuery
```

Comment:
I think my previous comment was lost in an [outdated conversation](https://github.com/alphagov/content-tagger/pull/519#discussion_r152553193) so I'm copying the comment I made in here as I still think it's applicable.

---

> TaxonomyQuery.new.taxonomy returns an Array of numbers - to have a 'depth' method I'd have to have it return an instance of a custom class, which seems a bit heavyweigth

It might be something we want to have, though perhaps I'm the only one that finds the class interface quite difficult to understand what going to be returned. 

Using one of the `taxonomy_query_spec.rb` examples, we seem to be returning arrays of hashes and then relying on Ruby's array methods to return the types of queries we're after.

```rb
taxonomy = TaxonomyQuery.new.taxonomy
=> [
  [
    {"content_id"=>"root_taxon_a", "base_path"=>"/taxons/root_taxon_a"},
    {"content_id"=>"root_taxon_b", "base_path"=>"/taxons/root_taxon_b"}
  ],
  [
    {"content_id"=>"child_a_1", "base_path"=>"/taxons/child_a_1"},
    {"content_id"=>"child_a_2", "base_path"=>"/taxons/child_a_2"},
    {"content_id"=>"child_b_1", "base_path"=>"/taxons/child_b_1"},
    {"content_id"=>"child_b_2", "base_path"=>"/taxons/child_b_2"}
  ]
]

taxonomy.size
=> 2

taxonomy.first
=> [
  {"content_id"=>"root_taxon_a", "base_path"=>"/taxons/root_taxon_a"},
  {"content_id"=>"root_taxon_b", "base_path"=>"/taxons/root_taxon_b"}
]

```

Perhaps renaming this class to something like `TaxonomyDepthQuery` that has public methods `depth`, `taxons_at_level(n)` might clarify how to use the query?

Reply:
I renamed this method to 'taxons per level' I hope that clarifies it a bit